### PR TITLE
Update NetTopologySuite to 2.0.0

### LINKED ISSUE #14825: Update to NetTopologySuite v2
This version of NetTopologySuite has some fairly significant breaking changes. (e.g. splitting Coordinate into types like CoordinateXY, CoordinateXYZ, etc.) If our timelines align, it would be nice to upgrade to the new version during our 3.0.0 release. If not, we may need to consider creating packages like Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite2 or something.

cc @airbreather @FObermaier

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #14825

cc @airbreather

### CODE REVIEW

File: src/EFCore.Sqlite.NTS/Storage/Internal/SqliteGeometryTypeMapping.cs [hunk 1]
```diff
@@ -126,12 +126,12 @@ protected override int GetSrid(object value)
...
 
-        private static GaiaGeoReader CreateReader(IGeometryServices geometryServices)
+        private static GaiaGeoReader CreateReader(NtsGeometryServices geometryServices)
             => new GaiaGeoReader(
                 geometryServices.DefaultCoordinateSequenceFactory,
                 geometryServices.DefaultPrecisionModel);
 
         private static GaiaGeoWriter CreateWriter()
-            => new GaiaGeoWriter();
+            => new GaiaGeoWriter { HandleOrdinates = Ordinates.XY };
```
Comment (bricelam) [inline src/EFCore.Sqlite.NTS/Storage/Internal/SqliteGeometryTypeMapping.cs]: Will address as part of #13767. Maintaining compatibility for now.

File: test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs [hunk 2]
```diff
@@ -2433,7 +2432,7 @@ public void InsertDataOperation_required_args()
                 "mb.InsertData(" + _eol +
                 "    table: \"People\"," + _eol +
                 "    column: \"Geometry\"," + _eol +
-                "    value: (NetTopologySuite.Geometries.Point)new NetTopologySuite.IO.WKTReader().Read(\"SRID=4326;POINT (1.1 2.2 3.3)\"));",
+                "    value: (NetTopologySuite.Geometries.Point)new NetTopologySuite.IO.WKTReader().Read(\"SRID=4326;POINT Z(1.1 2.2 3.3)\"));",
```
Comment (bricelam) [inline test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs]: @airbreather This is incompatible with SQL Server. Is there any way to go back to the previous format? Apparently we don't have a test covering it, but we use WKT for geometry literals/constants in the SQL and I suspect they're broken now.

Comment (airbreather) [reply test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs]: > This is incompatible with SQL Server

Confirmed, added NetTopologySuite/NetTopologySuite#324

Comment (airbreather) [reply test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs]: Should be fixed in 2.0.0-pre.180690659 now

Comment (bricelam) [reply test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs]: Wonderful, thank you!

Comment (bricelam) [reply test/EFCore.Design.Tests/Migrations/Design/CSharpMigrationOperationGeneratorTest.cs]: Works great; added test coverage.

**Comment by bricelam** [comment]
@airbreather Everything looks good on our end. If you publish a prerelease to NuGet sometime this week, we can get this into our July prerelease (Preview 7)

**Comment by bricelam** [comment]
(Windows CI failure unrelated--MyGet issue/outage)

**Pull request merged**